pyproject.toml

```diff
@@ -1,11 +1,41 @@
-[tool.black]
+[tool.ruff]
+# https://docs.astral.sh/ruff/configuration/
+target-version = "py38"
 line-length = 79
-exclude = '''
-(
-  \.git
-  | \.tox
-  | venv
-  | build
-  | dist
-)
-'''
+extend-exclude = [
+  "_template",
+  "*_pb2*.py*",
+]
+output-format = "concise"
+
+[tool.ruff.lint]
+# https://docs.astral.sh/ruff/linter/#rule-selection
+# pylint: https://github.com/astral-sh/ruff/issues/970
+select = [
+  "I",   # isort
+  "F",   # pyflakes
+  "E",   # pycodestyle errors
+  "W",   # pycodestyle warnings
+  "PLC", # pylint convention
+  "PLE", # pylint error
+  "Q",   # flake8-quotes
+  "A",   # flake8-builtins
+]
+ignore = [
+  "E501", # line-too-long
+]
+
+[tool.ruff.lint.per-file-ignores]
+"docs/**/*.*" = ["A001"]
+
+[tool.ruff.lint.isort]
+detect-same-package = false # to not consider instrumentation packages as first-party
```

Comment:
Had to add this parameter https://docs.astral.sh/ruff/settings/#lint_isort_detect-same-package
to tell ruff to avoid considering instrumentation packages as first-party. One other option is to use known-third-party with all packages we instrument (e.g, known-third-party = ["sqlalchemy", "httpx", ...]